sheerwater_benchmarking/utils/caching.py

```diff
@@ -444,10 +435,177 @@ def cache_exists(backend, cache_path, verify_path=None):
         raise ValueError(f'Unknown backend {backend}')
 
 
-def cacheable(data_type, cache_args, timeseries=None, chunking=None, chunk_by_arg=None,
-              auto_rechunk=False, cache=True, validate_cache_timeseries=True, cache_disable_if=None,
-              backend=None, storage_backend=None):
-    """Decorator for caching function results.
+def update_config_from_kwargs(cache_config, passed_kwargs):
+    """Extract the cache arguments from the kwargs and return them."""
+    for k in cache_config:
+        if k in passed_kwargs and passed_kwargs[k] is not None:
+            # Overwrite the cache config parameter with the passed kwargs
+            # if provided and is not None
+            cache_config[k] = passed_kwargs[k]
+            # Remove cache config parameter from the passed kwargs
+            del passed_kwargs[k]
+    return cache_config
+
+
+def get_cache_config(cache_config, func, args, kwargs):
+    """Get the runtime configuration of the cache.
+
```

Comment:
A ~copy of the code in the first half of our cacheable function - I wish it diffed better.